projects/fsastorefrontlib/src/cms-components/agent/appointment-scheduling-confirmation/appointment-scheduling-confirmation.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, OnInit } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import { RoutingService } from '@spartacus/core';
+import { genericIcons } from '../../../assets/icons/generic-icons';
+
+@Component({
+  selector: 'cx-fs-appointment-scheduling-confirmation',
+  templateUrl: 'appointment-scheduling-confirmation.component.html',
+})
+export class AppointmentSchedulingConfirmationComponent implements OnInit {
+  constructor(
+    protected domSanitizer: DomSanitizer,
+    protected routingService: RoutingService
+  ) {}
+
+  appointmentDate: string;
+
+  ngOnInit() {
+    if (history.state.date) {
+      this.appointmentDate = history.state.date;
+    } else {
+      this.routingService.go('/');
+    }
+  }
+
+  getImagelink() {
```

Comment:
I know this is inherited code from other implementations, but it should be refactored, i.e. deprecated. We should place this in a property (maybe just `imageLink`), and deprecate this implementation in other components, since like this it's being re-evaluated every time on any change detection. For the deprication, we can create a new ticket, and for this - just move it to a property.